ui/js/models/ph_series.js

```diff
@@ -0,0 +1,228 @@
+        var signatureURL = thServiceDomain + '/api/project/' + projectName +
+            '/performance/signatures/?interval=' +
+            timeRange;
+
+        return $http.get(signatureURL).then(function(response) {
+            var seriesList = [];
+            var platformList = [];
+            var testList = [];
+
+            Object.keys(response.data).forEach(function(signature) {
+                var seriesSummary = _getSeriesSummary(projectName, signature,
+                                                      response.data[signature],
+                                                      optionMap);
+
+                seriesList.push(seriesSummary);
+
+                // add test/platform to lists if not yet present
+                if (!_.contains(platformList, seriesSummary.platform)) {
+                    platformList.push(seriesSummary.platform);
+                }
+                if (!_.contains(testList, seriesSummary.name)) {
+                    testList.push(seriesSummary.name);
+                }
+            });
+            return {
+                seriesList: seriesList,
+                platformList: platformList,
+                testList: testList
+            };
+        });
+    };
+
+    var _getPlatformList = function(projectName, timeRange) {
+        var platformURL = thServiceDomain + '/api/project/' + projectName +
+            '/performance/platforms/?interval=' +
+            timeRange;
+        return $http.get(platformURL).then(function(response) {
+            return {
+                platformList: response.data
+            };
+        });
+
+    };
+
+    var _getSeriesByPlatform = function(projectName, timeRange, platform, optionMap) {
+        var specifyPlatformURL = thServiceDomain + '/api/project/' + projectName +
+            '/performance/signatures/?interval=' +
+            timeRange + '&platform=' + platform;
+
+        return $http.get(specifyPlatformURL).then(function(response) {
+            var seriesList = [];
+            var testList = [];
+
+            _.keys(response.data).forEach(function(signature){
+                var seriesSummary = _getSeriesSummary(projectName, signature,
+                    response.data[signature], optionMap);
+                seriesList.push(seriesSummary);
+
+                // add test/platform to lists if not yet present
+                if (!_.contains(testList, seriesSummary.name)) {
+                    testList.push(seriesSummary.name);
+                }
+            });
+
+            return {
+                platform: platform,
+                seriesList: seriesList,
+                testList: testList
+            };
+        });
+    };
+
+    var _getSeriesByJobId = function(projectName, jobId) {
+        var JobIdURL = thServiceDomain + '/api/project/' + projectName +
+            '/performance/data/?job_id=' + jobId;
+        return $http.get(JobIdURL).then(function(response) {
```

Comment:
It shouldn't be necessary to create a temporary variable called jobIdURL here -- just pass it directly to $http.get.

Also the first letter of a variable like this should be uncapitalized. I.e. `JobId` -> `jobId`